### ISSUE TITLE
Direct Tcp connectivity not working in emulator

### ISSUE THREAD

**Issue by JamesPSScott** [Azure/azure-documentdb-dotnet#400]
Hi all,

I did some Windows Updates yesterday (One was the Fall Creators update I think).  I also installed the latest VS 2017 15.5.  Since then Cosmos Db Emulator will not let me connect in Direct Mode.  I did figure out how to connect in gateway, but for testing reasons we really need to connect in both.

I disabled my firewall to see if that was the issue.  It wasn't.

Please help!!  I already lost a day to this unfortunately...

**Comment by kirankumarkolli** [comment]
@JamesPSScott which version of emulator are you using? can you please try with latest version.

**Comment by JamesPSScott** [comment]
Its 1.19.102.5.  I downloaded this on Monday from MS site.

**Comment by kirankumarkolli** [comment]
@JamesPSScott are you still facing the issue? 

Did you follow instructions at https://docs.microsoft.com/en-us/azure/cosmos-db/local-emulator#troubleshooting

**Comment by JamesPSScott** [comment]
Yes, I'm still experiencing the issue.  I converted all of my code over to not use direct TCP for now.  Unfortunately it was the only way I could proceed with nobody to help me.

**Comment by kirankumarkolli** [comment]
@JamesPSScott  did you send mail to askcosmosdb@microsoft.com with the details mentioned in above thread?

**Comment by JamesPSScott** [comment]
Well the last few times I sent anything to askcosmosdb@microsoft.com OR the docdb email before that I didn't get a response, so I wasn't sure it would work this time either.  LOL

When I have some time I'll see if I can get those trace files packed up and send them over.

Thanks!

**Comment by grahamehorner** [comment]
I’ve had the same experience with the email, not even a simple we got your email reply 😭

**Comment by JamesPSScott** [comment]
@grahamehorner well that's good.  I thought it was just me LOL

**Comment by kirankumarkolli** [comment]
@JamesPSScott apologies for your prior experiences. Please do try once more sending mail to askcosmosdb@microsoft.com with details and we will respond to it.

**Comment by mlankamp** [comment]
I had exactly the same problem when running the emulator in the office, while at home i could use a direct connection. I tried until I found the DocumentDbEmulatorCertificate (mmc.exe -> Add/Remove snap-in -> certificates -> Computer account). In the certificate the is a property Subject Alternative Name, in this property I found different DNS names but also my IP-adresses.
If I shutdown the emulator and delete this certificate (only this one, leave the DocDBEmulatorSecrets as is) and restart the emulator the certificate is recreated and a direct connection is possible.

**Comment by josiahdj** [comment]
@mlankamp Worked for me, thanks!

**Comment by AmitSuri** [comment]
TYVM!! @mlankamp 
@kirankumarkolli   : Please look at this solution. It works.

**Comment by filter-coffee** [comment]
@dotjoe 's solution worked for me as well. Thanks.

**Comment by tanwarsatya** [comment]
@mlankamp your solution worked. Look's like default certificate has some developer's  machine name who may be working on emulator.

**Comment by MartyLanda** [comment]
@mlankamp This worked for me as well. The emulator picked up on the ip address assigned to VirtualBox on the same machine. I had to set a metric value for the the primary network connection so the emulator would find that first when re-creating the certificate.
